sourcegraph/sourcegraph
# web: trace search results and search query input

## Context

Produce OpenTelemetry spans by several key React components:

1. `CodeMirrorQueryInput`
2. `StreamingSearchResultsListItem`
3. `StreamingSearchResults`
4. `SearchPage`
5. `SearchBox`

Closes https://github.com/sourcegraph/sourcegraph/issues/40468

## Test plan

1. `ENABLE_OPEN_TELEMETRY=true sg start web-standalone`
2. Open the browser console and ensure that debug-level output is enabled.
3. Open the search results page.
4. See spans created by the components listed above.

<img width="561" alt="Screenshot 2022-11-09 at 11 32 45" src="https://user-images.githubusercontent.com/3846380/200731833-15d29439-c52f-4ae9-92b5-9beac4cf2774.png">


## App preview:

- [Web](https://sg-web-vb-trace-more-react-components.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
